copycat-api/src/main/java/org/apache/kafka/copycat/source/SourceConnector.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package org.apache.kafka.copycat.source;
+
+import org.apache.kafka.copycat.connector.Connector;
+
+/**
+ * SourceConnectors implement the connector interface to pull data from another system and send
+ * it to Kafka.
+ */
+public abstract class SourceConnector extends Connector {
```

Comment:
I'm not quite sure. One option would be to require SourceConnectors/SourceTasks to implement a method to get this information. However, users of the embedded API wouldn't be creating those classes themselves, just submitting a job, so they couldn't call this directly. I think we'd want chain it through other APIs all the way up to whatever top-level class is instantiated for embedded mode (i.e. the Copycat class that you would call run() and addConnector() on).